packages/voting-contract/src/lib.rs

```diff
@@ -172,10 +172,19 @@ where
 
     // Set it to executed
     proposal.status = Status::Executed;
-    proposals::<P>().save(deps.storage, proposal_id, &proposal)?;
+    proposals::<P>().save(storage, proposal_id, &proposal)?;
     Ok(proposal)
 }
 
+pub fn execute_text<P>(deps: DepsMut, id: u64, proposal: &Proposal<P>) -> Result<(), ContractError>
```

Comment:
Here I would take `&Proposal<P>` as input, or even `Proposal<P>`, but convert it into `ProposalInfo` with the same data everywhere (title and description) and save that. Normalizing before saving make us clear what data we actually have in the db.